tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -620,3 +621,40 @@ def test_daemon_auto_shutdown_after_first_run(self):
 
       for process in pantsd_runner_processes:
         self.assertFalse(process.is_running())
+
+  def test_dependencies_swap(self):
```

Comment:
Could we add a very brief comment noting that this tests that the graph is left in a consistent state (without cycles) after swapping? It's not clear what this is guarding against (and therefore not clear how to extend).